### PR TITLE
Add button to clear output view

### PR DESCRIPTION
**Pull request by simark**
This patch adds a button to clear the currently displayed output
channel.

Change-Id: I346b7b3bc5f7fcee6f4fed31c3390ad6847a2d53

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
> The button and drop-down are not perfectly aligned (the button is 1 px lower), but I can't get the css right to get them aligned.

I was messing around with it a little and got to this point. Maybe the real noticeable difference is made by removing the background color to match that of the `Search Widget`.

selection_106

**Comment by simark** [comment]
> I was messing around with it a little and got to this point. Maybe the real noticeable difference is made by removing the background and border color to match that of the `Search Widget`.

Yeah. if there's no background, I guess you can't notice it's not aligned :)

**Comment by jbicker** [comment]
Try to set the `display` value of `outputOverlay` container to `flex`. :-)

**Comment by vince-fugnitto** [comment]
> Try to set the `display` value of `outputOverlay` container to `flex`. :-)
> screen shot 2018-10-18 at 15 36 36

I tried that aswell, but if you notice the background on the `clear` the padding on the left is larger than that of the right.

**Comment by simark** [comment]
Thanks @jbicker, it helps.

> I tried that aswell, but if you notice the background on the `clear` button the padding on the left is larger than that of the right.

It's the same as my original screenshot, so I don't think it's caused by the flex.  This PR is ready for review :)

File: packages/output/src/browser/output-widget.tsx [hunk 1]
```diff
@@ -88,7 +88,27 @@ export class OutputWidget extends ReactWidget {
...
+        return <React.Fragment>
+            <div id='outputOverlay'>
+                {this.renderChannelSelector()}
+                {this.renderClearButton()}
+            </div>
+            {this.renderChannelContents()}
+        </React.Fragment>;
+    }
+
+    private readonly OUTPUT_CLEAR_BUTTON_ID = 'outputClear';
```
Comment (kittaakos) [inline packages/output/src/browser/output-widget.tsx]: Let's have a `const` somewhere for the CSS classes, or use `static` fields instead. Can you please update the other classes too? We [did this](https://github.com/theia-ide/theia/blob/e619393d725bbf5a05b260169193c7c538e55e4c/packages/git/src/browser/git-widget.tsx#L653-L669) in other widgets.

File: packages/output/src/browser/output-widget.tsx [hunk 2]
```diff
@@ -88,7 +88,27 @@ export class OutputWidget extends ReactWidget {
     }
 
     protected render(): React.ReactNode {
-        return <React.Fragment>{this.renderChannelSelector()}{this.renderChannelContents()}</React.Fragment>;
+        return <React.Fragment>
+            <div id='outputOverlay'>
```
Comment (kittaakos) [inline packages/output/src/browser/output-widget.tsx]: Please use a constant for the ID.

File: packages/output/src/browser/output-widget.tsx [hunk 3]
```diff
@@ -88,7 +88,27 @@ export class OutputWidget extends ReactWidget {
...
+
+    private readonly OUTPUT_CLEAR_BUTTON_ID = 'outputClear';
+
+    protected renderClearButton(): React.ReactNode {
+        return <span title='Clear'
+            className={this.selectedChannel ? 'enabled' : ''}
+            id={this.OUTPUT_CLEAR_BUTTON_ID} onClick={() => this.clear()} />;
+    }
+
+    private clear() {
```
[the remark this answers is not in the document]
Comment (simark) [reply packages/output/src/browser/output-widget.tsx]: Done.  I also prefer explicit return types.  If there is a linter rule to enforce this, would you be for enabling it?

File: packages/output/src/browser/output-widget.tsx (hunk 1 above)
Comment (simark) [reply packages/output/src/browser/output-widget.tsx]: This is an id, not a css class, but I can certainly extract it in the same fashion.

**Comment by simark** [comment]
I have pushed the patch, but if you'd like me to do anything differently please leave comments, and I'll do it as a separate PR.

**Pull request merged**